internal/operator-controller/applier/boxcutter.go

```diff
@@ -135,6 +140,54 @@ func (r *SimpleRevisionGenerator) GenerateRevision(
 	return r.buildClusterExtensionRevision(objs, ext, revisionAnnotations), nil
 }
 
+var restrictedMetadata = []string{
+	"finalizers",
+	"ownerReferences",
+	"creationTimestamp",
+	"uid",
+	"resourceVersion",
+	"generation",
+	"managedFields",
+	"deletionTimestamp",
+	"deletionGracePeriodSeconds",
+}
+
+// stripMetadata takes an unstructured obj and removes any of the above listed fields in-place, if they are present.
+// If any of the above fields are found and removed, a warning will be logged.
+func stripMetadata(ctx context.Context, obj *unstructured.Unstructured) {
+	l := log.FromContext(ctx)
+	m := obj.Object
+	strippedFields := []string{}
+
+	// remove status
+	if status, ok := m["status"].(map[string]any); ok && len(status) > 0 {
+		strippedFields = append(strippedFields, "status")
+		delete(m, "status")
+	}
+
+	var metadata map[string]any
+	if metaRaw, ok := m["metadata"]; ok {
+		metadata, ok = metaRaw.(map[string]any)
+		if !ok {
+			return
```

Comment:
```suggestion
            l.Info("warning: unexpected metadata type in manifest, skipping sanitization",
             "kind", obj.GetKind(),
             "name", obj.GetName(),
             "type", fmt.Sprintf("%T", metaRaw))
			return
```

Reply:
I like the idea but drilling down into `obj.GetKind()` and `obj.GetName()` it looks like eventually they perform the exact same type-cast, and return `""` when it fails. Therefore if we fail the type-cast the `kind` and `name` in this log will always be empty strings - assuming I've interpreted it correctly ([reference](https://github.com/kubernetes/apimachinery/blob/b72d93d174332f952a8d431419fece5e6f044bcb/pkg/apis/meta/v1/unstructured/helpers.go#L61)). We could log regardless, but if we can't pull any identifiers will it still be helpful?

Reply:
Yup that makes sense, I don't think the log message will be helpful if we aren't pulling any identifiers. 
Thanks for clarifying @dtfranz!